app/forms/training_details_form.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class TrainingDetailsForm
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+
+  attr_accessor :trainee, :trainee_id, :day, :month, :year, :validate_commencement_date, :validate_trainee_id
+
+  delegate :id, :persisted?, to: :trainee
+
+  validates :trainee_id, presence: true, if: -> { validate_trainee_id }
+
+  validate :commencement_date_valid, if: -> { validate_commencement_date }
```

Comment:
Did we come to a conclusion?

> 2000+ years ago?
> 
> ![image](https://user-images.githubusercontent.com/470137/107927907-8ba79200-6f6f-11eb-8495-c7e0cdcd03ac.png)

Reply:
@defong Nick said a valid date will suffice. I guess we can create another card later on if necessary.

Reply:
I'd probably match the programme start date validation. If not for this card, then we can raise another card.